### PR TITLE
[Bugfix][Model Runner V2] Preserve all allowed_token_ids in the logit bias kernel

### PR DESCRIPTION
**Pull request by Sunt-ing**
## Purpose

Under the V2 model runner, `SamplingParams.allowed_token_ids` can silently drop some of the allowed tokens. Generation then produces a token outside the allowlist, and when every allowed token happens to be dropped the whole logits row becomes `-inf` and the engine samples token id `0`.

Root cause is in the Triton `_bias_kernel` in `vllm/v1/worker/gpu/sample/logit_bias.py`. It applies the allowlist in three steps: (1) load the current logits for the allowed ids into registers, (2) overwrite the entire row with `-inf`, (3) write the saved logits back to the allowed positions. Steps (1) and (2) access the same logit addresses from different lanes of the same kernel with no barrier between them, so for some allowed ids the `-inf` overwrite races ahead of the save: the save reads `-inf`, and step (3) writes `-inf` back, dropping that allowed token. Which ids are lost depends on scheduling; on a given build most ids are consistently dropped or kept while a few are borderline across runs.

The fix adds two `tl.debug_barrier()` calls (already used in `causal_conv1d.py` for the same class of Triton ordering hazard): one so every save completes before the `-inf` overwrite, one so the overwrite completes before the restore.

## Test Plan

- Extend the existing `tests/v1/sample/test_sampling_params_e2e.py::test_allowed_token_ids` with single-token allowlists spread across the vocab; each must force exactly that token (the kernel used to drop some).
- Black-box end-to-end check on a public model (`Qwen/Qwen3-0.6B`): a restricted `allowed_token_ids` request must only ever sample allowed tokens. Reproduction script below.

## Test Result

On `main` a restricted `allowed_token_ids` request samples tokens outside the allowlist; with this fix every allowed token is preserved. 902's logit (9.31) outranks 9834's (8.06), so the dropped token is often the true argmax, which changes the sampled output.

<details>
<summary>End-to-end reproduction (environment + script + output)</summary>

**Environment**
- GPU: NVIDIA RTX 4090
- vLLM: `main` at commit `5099474`
- Model: `Qwen/Qwen3-0.6B`, `enforce_eager=True`, greedy (`temperature=0`)

**Script**
```python
from vllm import LLM, SamplingParams

llm = LLM("Qwen/Qwen3-0.6B", enforce_eager=True)
for allowed in ([902], [6303, 2518, 902, 9834]):  # 902 == " no", 9834 == " yes"
    out = llm.generate(
        ["Write one answer token:"],
        SamplingParams(temperature=0, max_tokens=1, allowed_token_ids=allowed),
    )[0].outputs[0]
    print(allowed, "->", out.token_ids[-1], repr(out.text))
```

**Output**
```
# main
[902] -> 0 '!'                          # 0 is not in allowed_token_ids
[6303, 2518, 902, 9834] -> 9834 ' yes'  # 902 and 2518 dropped

# this PR
[902] -> 902 ' no'
[6303, 2518, 902, 9834] -> 902 ' no'
```

</details>

The existing `tests/v1/sample/test_sampling_params_e2e.py::test_allowed_token_ids`, extended to sweep ids across the vocab, fails on the unmodified kernel and passes with the barriers; the full file (allowed_token_ids, bad_words, stop_token_ids, penalties, min_tokens, seed, n>1) is 8 passed.

AI assistance was used to investigate, reproduce, and draft this change; the author reviewed the diff and validation output.

### CODE REVIEW

**Review by WoosukKwon — APPROVED** [review]
LGTM! thanks for catching the bug.

**Comment by AndreasKaratzas** [comment]
PR for that regression is here: 
- https://github.com/vllm-project/vllm/pull/46401
